test/unit/models/classes/runner/TestDefinitionSerializerServiceTest.php

```diff
@@ -115,5 +123,29 @@ public function testGetSerializedTestDefinition()
                 ],
             ], $this->testDefinitionSerializerService->getSerializedTestDefinition($this->qtiRunnerServiceContext));
         }
+
+        /**
+         * @return ResourceFactory|\PHPUnit_Framework_MockObject_MockObject
+         */
+        private function getResourceFactoryMock()
+        {
+            $resourceFactoryMock = $this
+                ->getMockBuilder(ResourceFactory::class)
+                ->getMock();
+
+            $resourceFactoryMock
+                ->method('create')
+                ->will($this->returnCallback([$this, 'resourceFactoryCreateCallback']));
```

Comment:
You can define the callback as an inline function, it's better imho. What do you think?